src/services/rfqm_service.ts

```diff
@@ -0,0 +1,138 @@
+import { MarketOperation, QuoteRequestor } from '@0x/asset-swapper';
+
+import { RFQT_REQUEST_MAX_RESPONSE_MS } from '../config';
+import { NULL_ADDRESS } from '../constants';
+import {
+    FetchFirmQuoteParams,
+    FetchFirmQuoteResponse,
+    FetchIndicativeQuoteParams,
+    FetchIndicativeQuoteResponse,
+} from '../types';
+
+/**
+ * RfqmService is the coordination layer for HTTP based RFQM flows
+ */
+export class RfqmService {
+    constructor(private readonly _quoteRequestor: QuoteRequestor) {}
+
+    /**
+     * Fetch the best indicative quote available
+     */
+    public async fetchIndicativeQuoteAsync(params: FetchIndicativeQuoteParams): Promise<FetchIndicativeQuoteResponse> {
+        const { sellAmount, buyAmount, sellToken, buyToken, apiKey } = params;
```

Comment:
Can we use a single `amount` param instead of `buyAmount` and `sellAmount`?

Reply:
Good idea - I mostly copied over the params from the swap endpoint, but better to start simple and add on this complexity if we really need it